Work In Progress: CI script for Linux and OSX

cdmihai opened the pull request:
- Added shell script for Linux and OSX. It has two scopes, compilation and testing.
- Updated netci.groovy script to account for Linux and OSX. Linux is currently switched off.

@mmitche 

Items left:
- [x] update cibuild.sh to download zip with mono build of msbuild

mmitche commented on cibuild.sh:
```diff
@@ -0,0 +1,75 @@
+usage()
+{
+    echo "Options"
+    echo "  --system-msbuild-path <path>   Path to the mono msbuild build used to build msbuild" 
```
By this do you mean mono's xbuild or a mono hosted MSBuild?

cdmihai replied:
A mono hosted MSBuild. I'll update the comment.

mmitche commented on cibuild.sh:
```diff
@@ -0,0 +1,75 @@
+{
+    echo "Options"
+    echo "  --system-msbuild-path <path>   Path to the mono msbuild build used to build msbuild" 
+    echo "  --os <os>                      OS to run (Linux / Darwin)"
```
This can be detected by default using uname.  Check out the run-test.sh script in CoreFx for a good example.

cdmihai commented:
@mmitche I implemented your feedback and also modified cibuild.sh to download the mono hosted msbuild binaries.

Everything should be ready now for a CI run :) Can you please check if everything looks good?

mmitche commented:
@dotnet-bot test ci please

mmitche commented on netci.groovy:
```diff
@@ -6,38 +6,60 @@ def project = GithubProject
+        // Define job.
+        switch(osName) {
+            case 'Windows_NT':
+                myJob.with{
```
newJob.with {

mmitche commented on netci.groovy:
```diff
@@ -6,38 +6,60 @@ def project = GithubProject
+                }
+                break;
+            case 'OSX':
+                myJob.with{
```
newJob.with {

mmitche commented on netci.groovy:
```diff
@@ -6,38 +6,60 @@ def project = GithubProject
+                }
+                break;
+            case 'Ubuntu':
+                myJob.with{
```
newJob.with {

cdmihai commented:
@dotnet-bot test ci please

cdmihai commented:
@dotnet-bot test ci please

cdmihai commented:
@dotnet-bot test ci please

mmitche commented:
@cdmihai The ci test phrase is designed to check that the jobs can be generated.  Once merged, the jobs will start to run automatically on the desired schedules.

cdmihai commented:
@dotnet-bot test ci please

cdmihai commented:
@dotnet-bot test this please

mmitche commented:
Looks correct to me.  Feel free to merge.

rainersigwald commented:
LGTM
